GitUI/CommandsDialogs/RevisionDiffController.cs

```diff
@@ -121,8 +124,7 @@ public bool ShouldShowMenuEditWorkingDirectoryFile(ContextMenuSelectionInfo sele
 
         public bool ShouldShowMenuDeleteFile(ContextMenuSelectionInfo selectionInfo)
         {
-            return selectionInfo.IsSingleGitItemSelected && !selectionInfo.IsAnySubmodule &&
-                   selectionInfo.SingleFileExists && selectionInfo.SelectedRevision.IsArtificial;
+            return selectionInfo.IsAnyFileExist && selectionInfo.SelectedRevision.IsArtificial;
```

Comment:
IsArtificial should be working tree only (no new issue - Index is handled but not entirely logical)

The option should not be active when there are items that cannot be deleted. Maybe OK to check if there are items that cannot be deleted?

            return !selectionInfo.IsAnySubmodule && selectionInfo.SelectedRevision.IsAnyItemWorkTree ;